src/Transport/QueueTransportFactory.php

```diff
@@ -26,4 +38,48 @@ public function supports(string $dsn, array $options): bool
     {
         return 0 === strpos($dsn, 'azurequeue://');
     }
+
+    /**
+     * @param string $dsn
+     * @param array $options
+     *
+     * @return array
+     */
+    private function computeOptions(string $dsn, array $options): array
+    {
+        $resultOptions = $options;
+
+        if (false === $parsedUrl = parse_url($dsn)) {
+            throw new InvalidArgumentException('The given DSN is invalid.');
+        }
+
+        if (isset($parsedUrl['query'])) {
+            parse_str($parsedUrl['query'], $dsnOptions);
+            $resultOptions = array_merge($options, $dsnOptions);
+        }
+
+        return $this->castOptionsToTheRightType($resultOptions);
+    }
+
+    /**
+     * @param array $options
+     *
+     * @return array
+     */
+    private function castOptionsToTheRightType(array $options): array
+    {
+        foreach ($options as $key => $value) {
+            if (!isset(self::OPTIONS_TYPE_MAPPING[$key])) {
+                continue;
+            }
+
+            $options[$key] = match (self::OPTIONS_TYPE_MAPPING[$key]) {
```

Comment:
`match()` was introduced in PHP 8.0 but the package is compatible with PHP >= 7.3

Reply:
done, it was replaced with "switch"